### PR TITLE
Cli adaptation

### PR DESCRIPTION
**Pull request by ehoelzl**
This PR adds the possibility to chose the backend when running a benchmark task.

Linked to [#35](https://github.com/mlbench/mlbench-benchmarks/pull/35) in `mlbench-benchmarks`
and  [#22](https://github.com/mlbench/mlbench-dashboard/pull/22) in `mlbench-dashboard`

### CODE REVIEW

File: mlbench_core/api.py [hunk 1]
```diff
@@ -409,9 +394,11 @@ def create_run(
         num_cpus=2.0,
         max_bandwidth=1000,
         image=None,
+        backend=None,
         custom_image_name=None,
         custom_image_command=None,
-        custom_image_all_nodes=False,
+        custom_backend=None,
+        run_all_nodes=False,
```
Comment (Panaetius) [inline mlbench_core/api.py:401]: I think the default should be `None` and having both this and a non custom backend specified should raise an exception that they're mutually exclusive.

File: mlbench_core/api.py [hunk 2]
```diff
@@ -452,21 +447,28 @@ def create_run(
...
             "light_target": "true" if light_target else "false",
+            "run_all_nodes": "true" if run_all_nodes else "false",
         }
 
         if custom_image_name is not None:
             data["image_name"] = "custom_image"
             data["custom_image_name"] = custom_image_name
             data["custom_image_command"] = custom_image_command
-            data["custom_image_all_nodes"] = custom_image_all_nodes
+            # data["custom_image_all_nodes"] = custom_image_all_nodes
```
Comment (Panaetius) [inline mlbench_core/api.py:457]: Delete this if it's not needed, no need to polute the repo with commented out code, that's what git is for

File: mlbench_core/cli/cli.py [hunk 3]
```diff
@@ -130,17 +130,37 @@ def run(name, num_workers, gpu, light, dashboard_url):
         # run custom image
         image = click.prompt("Image", type=str)
         image_command = click.prompt("Command", type=str)
-        run_on_all = click.confirm(
-            "Run command on all nodes (otherwise just first node)"
-        )
+        # run_on_all = click.confirm(
```
Comment (Panaetius) [inline mlbench_core/cli/cli.py:133]: see above

File: mlbench_core/cli/cli.py [hunk 4]
```diff
@@ -130,17 +130,37 @@ def run(name, num_workers, gpu, light, dashboard_url):
...
         benchmark = {"image": images[selection]}
 
+    # Backend Prompt
+    text_prompt = "Backend: \n\n"
+    text_prompt += "\n".join("[{}]\t{}".format(i, t) for i, t in enumerate(MLBENCH_BACKENDS))
+    text_prompt += "\n[{}]\tCustom Backend".format(len(MLBENCH_BACKENDS))
+    text_prompt += "\n\nSelection"
+
+    selection = click.prompt(
+        text_prompt, type=click.IntRange(0, len(MLBENCH_BACKENDS)), default=0
```
Comment (Panaetius) [inline mlbench_core/cli/cli.py:151]: Something i've thought about (also for the other prompts), it'd be awesome if the cli remembered the last input as default. If you'd like you can add this in this PR or I can make a new issue for it.

File: mlbench_core/api.py (hunk 1 above)
Comment (ehoelzl) [reply mlbench_core/api.py:401]: Do you mean for `run_all_nodes` ? or `backend`

File: mlbench_core/cli/cli.py (hunk 4 above)
Comment (ehoelzl) [reply mlbench_core/cli/cli.py:147]: So it would remember what arguments were used the last time the command was run ?

**Pull request merged**